One more `org.cqfn.diktat.ruleset.utils.KotlinParseException: Text is not valid`

```
org.cqfn.diktat.ruleset.utils.KotlinParseException: Text is not valid: [""${window.location}/execution/$id" + if (status != null) "?status=$status" else """]
```
when trying to format
```kotlin
    private fun getHrefToExecution(id: Long, status: TestResultStatus?) =
        "${window.location}/execution/$id" + if (status != null) "?status=$status" else ""
```

Full stacktrace:
```
[ERROR] 2021-12-03 18:42:55 Internal error has occurred in rule [string-concatenation]. Please make an issue on this bug at https://github.com/cqfn/diKTat/. 
                       As a workaround you can disable these inspections in yml config: <[STRING_CONCATENATION]>.
                       Root cause of the problem is in [D:\projects\save-cloud\save-frontend\src\main\kotlin\org\cqfn\save\frontend\components\views\HistoryView.kt] file.
org.cqfn.diktat.ruleset.utils.KotlinParseException: Text is not valid: [""${window.location}/execution/$id" + if (status != null) "?status=$status" else """]
	at org.cqfn.diktat.ruleset.utils.KotlinParser.createNode(KotlinParser.kt:75)
	at org.cqfn.diktat.ruleset.utils.KotlinParser.createNode$default(KotlinParser.kt:75)
	at org.cqfn.diktat.ruleset.rules.chapter3.StringConcatenationRule.fixBinaryExpressionWithConcatenation(StringConcatenationRule.kt:115)
	at org.cqfn.diktat.ruleset.rules.chapter3.StringConcatenationRule.access$fixBinaryExpressionWithConcatenation(StringConcatenationRule.kt:31)
	at org.cqfn.diktat.ruleset.rules.chapter3.StringConcatenationRule$detectStringConcatenation$1.invoke(StringConcatenationRule.kt:65)
	at org.cqfn.diktat.ruleset.rules.chapter3.StringConcatenationRule$detectStringConcatenation$1.invoke(StringConcatenationRule.kt:61)
	at org.cqfn.diktat.ruleset.constants.Warnings.warnAndFix(Warnings.kt:260)
	at org.cqfn.diktat.ruleset.constants.Warnings.warnAndFix$default(Warnings.kt:202)
	at org.cqfn.diktat.ruleset.rules.chapter3.StringConcatenationRule.detectStringConcatenation(StringConcatenationRule.kt:61)
	at org.cqfn.diktat.ruleset.rules.chapter3.StringConcatenationRule.logic(StringConcatenationRule.kt:44)
...
```

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/StringConcatenationRule.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/StringConcatenationRule.kt
@@ -15,10 +15,9 @@ import org.jetbrains.kotlin.psi.KtBinaryExpression
 import org.jetbrains.kotlin.psi.KtCallExpression
 import org.jetbrains.kotlin.psi.KtConstantExpression
 import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
-import org.jetbrains.kotlin.psi.KtOperationExpression
+import org.jetbrains.kotlin.psi.KtExpression
 import org.jetbrains.kotlin.psi.KtParenthesizedExpression
 import org.jetbrains.kotlin.psi.KtReferenceExpression
-import org.jetbrains.kotlin.psi.KtSafeQualifiedExpression
 import org.jetbrains.kotlin.psi.KtStringTemplateExpression
 
 /**
@@ -154,13 +153,14 @@ class StringConcatenationRule(configRules: List<RulesConfig>) : DiktatRule(
             // =========== "a " + "b" -> "a b"
             val rvalueTextNew = rvalueText?.trim('"')
             return "$lvalueText$rvalueTextNew"
-        } else if (binaryExpressionPsi.isRvalueCallExpression() || binaryExpressionPsi.isRvalueDotQualifiedExpression() ||
-                binaryExpressionPsi.isRvalueOperation() || binaryExpressionPsi.isRvalueSafeQualifiedExpression()) {
+        } else if (binaryExpressionPsi.isRvalueCallExpression()) {
             // ===========  "a " + foo() -> "a ${foo()}}"
             return "$lvalueText\${$rvalueText}"
         } else if (binaryExpressionPsi.isRvalueReferenceExpression()) {
             // ===========  "a " + b -> "a $b"
             return "$lvalueText$$rvalueText"
+        } else if (!binaryExpressionPsi.isRvalueParenthesized() && binaryExpressionPsi.isRvalueExpression()) {
+            return "$lvalueText\${$rvalueText}"
         } else if (binaryExpressionPsi.isRvalueParenthesized()) {
             val binExpression = binaryExpressionPsi.right?.children?.first()
             if (binExpression is KtBinaryExpression) {
@@ -198,14 +198,8 @@ class StringConcatenationRule(configRules: List<RulesConfig>) : DiktatRule(
     private fun KtBinaryExpression.isRvalueReferenceExpression() =
             this.right is KtReferenceExpression
 
-    private fun KtBinaryExpression.isRvalueDotQualifiedExpression() =
-            this.right is KtDotQualifiedExpression
-
     private fun KtBinaryExpression.isRvalueParenthesized() =
             this.right is KtParenthesizedExpression
-
-    private fun KtBinaryExpression.isRvalueOperation() =
-            this.right is KtOperationExpression
 
     private fun KtBinaryExpression.isLvalueDotQualifiedExpression() =
             this.left is KtDotQualifiedExpression
@@ -219,6 +213,6 @@ class StringConcatenationRule(configRules: List<RulesConfig>) : DiktatRule(
     private fun KtBinaryExpression.isLvalueConstantExpression() =
             this.left is KtConstantExpression
 
-    private fun KtBinaryExpression.isRvalueSafeQualifiedExpression() =
-            this.right is KtSafeQualifiedExpression
+    private fun KtBinaryExpression.isRvalueExpression() =
+            this.right is KtExpression
 }

--- a/diktat-rules/src/test/resources/test/paragraph3/string_concatenation/StringConcatenationExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/string_concatenation/StringConcatenationExpected.kt
@@ -33,3 +33,4 @@ val myTest22 = "string${foo()}"
 val myTest23 = x.toString() + foo()
 val myTest24 = foo() + "string"
 val myTest25 = "String ${valueStr?.value}"
+val myTest26 = "my string ${if (true) "1" else "2"}"

--- a/diktat-rules/src/test/resources/test/paragraph3/string_concatenation/StringConcatenationTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/string_concatenation/StringConcatenationTest.kt
@@ -33,3 +33,4 @@ val myTest22 = "string" + foo()
 val myTest23 = x.toString() + foo()
 val myTest24 = foo() + "string"
 val myTest25 = "String " + valueStr?.value
+val myTest26 = "my string " + if (true) "1" else "2"

